networkx/algorithms/coloring/tests/test_coloring.py

```diff
@@ -119,8 +119,16 @@ def test_is_equitable(self):
     def test_num_colors(self):
         G = nx.Graph()
         G.add_edges_from([(0, 1), (0, 2), (0, 3)])
+        coloring = {0: 0, 1: 1, 2: 2, 3: 3}
+        assert is_equitable(G, coloring, num_colors=5)
+        assert is_equitable(G, coloring)
         pytest.raises(nx.NetworkXAlgorithmError, nx.coloring.equitable_color, G, 2)
 
+        G = nx.Graph()
+        G.add_edges_from([(0, 1), (0, 2), (0, 3), (0, 4)])
+        coloring1 = {0: 1, 1: 1, 2: 2, 3: 2, 4: 2}
+        assert not is_equitable(G, coloring1, num_colors=6)
```

Comment:
These tests should probably go in a new function as `test_num_colors` is being used to test ` nx.coloring.equitable_color` (maybe we should have a better name). But could you move these tests to a new function or maybe add them to `test_is_equitable`. Thanks!